stackslib/src/chainstate/stacks/boot/pox_4_tests.rs

```diff
@@ -1234,3 +1250,405 @@ fn balances_from_keys(
         .map(|principal| get_stx_account_at(peer, tip, &principal))
         .collect()
 }
+
+#[test]
+fn stack_stx_signer_key() {
+    let lock_period = 2;
+    let (burnchain, mut peer, keys, latest_block, block_height, mut coinbase_nonce) =
+        prepare_pox4_test(function_name!());
+
+    let stacker_nonce = 0;
+    let stacker_key = &keys[0];
+    let min_ustx = get_stacking_minimum(&mut peer, &latest_block);
+
+    // (define-public (stack-stx (amount-ustx uint)
+    //                       (pox-addr (tuple (version (buff 1)) (hashbytes (buff 32))))
+    //                       (start-burn-ht uint)
+    //                       (lock-period uint)
+    //                       (signer-key (buff 33)))
+    let pox_addr = make_pox_addr(
+        AddressHashMode::SerializeP2WSH,
+        key_to_stacks_addr(stacker_key).bytes,
+    );
+    let signer_key_val = Value::buff_from(vec![
+        0x03, 0xa0, 0xf9, 0x81, 0x8e, 0xa8, 0xc1, 0x4a, 0x82, 0x7b, 0xb1, 0x44, 0xae, 0xc9, 0xcf,
+        0xba, 0xeb, 0xa2, 0x25, 0xaf, 0x22, 0xbe, 0x18, 0xed, 0x78, 0xa2, 0xf2, 0x98, 0x10, 0x6f,
+        0x4e, 0x28, 0x1b,
+    ])
+    .unwrap();
+    let txs = vec![make_pox_4_contract_call(
+        stacker_key,
+        stacker_nonce,
+        "stack-stx",
+        vec![
+            Value::UInt(min_ustx),
+            pox_addr,
+            Value::UInt(block_height as u128),
+            Value::UInt(2),
+            signer_key_val.clone(),
+        ],
+    )];
+
+    let latest_block = peer.tenure_with_txs(&txs, &mut coinbase_nonce);
+    let stacking_state = get_stacking_state_pox_4(
+        &mut peer,
+        &latest_block,
+        &key_to_stacks_addr(stacker_key).to_account_principal(),
+    )
+    .expect("No stacking state, stack-stx failed")
+    .expect_tuple();
+
+    let state_signer_key = stacking_state.get("signer-key").unwrap();
+    assert_eq!(state_signer_key.to_string(), signer_key_val.to_string());
+}
+
+#[test]
+fn delegate_stx_signer_key() {
```

Comment:
We need a test for `delegate-stack-stx`, which has a required signing key argument.  But this test should be dropped.

Reply:
Ah, I see you have one below.  But since `delegate-stx` doesn't take a singing key, this test can be dropped.